### PR TITLE
Enhanced RDoc for Struct

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Treats:
- #values_at
- #select
- #==
- #hash
- #eql?
- #size
- #dig

### CODE REVIEW

File: struct.c [hunk 1]
```diff
@@ -1433,14 +1487,15 @@ rb_struct_eql(VALUE s, VALUE s2)
 
 /*
  *  call-seq:
- *     struct.length    -> integer
- *     struct.size      -> integer
+ *    size -> integer
  *
- *  Returns the number of struct members.
+ *  Returns the count of members.
```
Comment (jeremyevans) [inline struct.c:1492]: I think `number` is better than `count` here.  `count` should be used when something requires an action per thing being counted, whereas the size of a struct is always fixed per class.

File: struct.c [hunk 2]
```diff
@@ -1295,18 +1316,27 @@ rb_struct_values_at(int argc, VALUE *argv, VALUE s)
...
- *     struct.select                 -> enumerator
- *     struct.filter {|obj| block }  -> array
- *     struct.filter                 -> enumerator
+ *    select {|value| ... } -> array
+ *    select -> enumerator
  *
- *  Yields each member value from the struct to the block and returns an Array
- *  containing the member values from the +struct+ for which the given block
- *  returns a true value (equivalent to Enumerable#select).
+ *  With a block given:
```
Comment (jeremyevans) [inline struct.c:1322]: This explanation is too long winded and goes into too much implementation detail. Maybe instead: `With a block given, returns an array of values from +self+ for which the block returns a truthy value`

File: struct.c [hunk 3]
```diff
@@ -1275,15 +1275,36 @@ struct_entry(VALUE s, long n)
...
+ *
+ *    Customer = Struct.new(:name, :address, :zip)
+ *    joe = Customer.new("Joe Smith", "123 Maple, Anytown NC", 12345)
+ *    joe.values_at(0, 2)    # => ["Joe Smith", 12345]
+ *    joe.values_at(2, 0)    # => [12345, "Joe Smith"]
+ *    joe.values_at(2, 1, 0) # => [12345, "123 Maple, Anytown NC", "Joe Smith"]
+ *    joe.values_at(0, -3)   # => ["Joe Smith", "Joe Smith"]
+ *
+ *  Raises IndexError if any of +integers+ is out of range.
+ *  Raises TypeError if any of +integers+ is not an integer.
```
Comment (jeremyevans) [inline struct.c:1294]: You can remove the TypeError line, as explained in the documentation guide section about exceptions.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Thanks for working on this.  Only a few requested changes.

File: struct.c [hunk 4]
```diff
@@ -1411,11 +1457,19 @@ recursive_eql(VALUE s, VALUE s2, int recur)
 
 /*
  * call-seq:
- *   struct.eql?(other)   -> true or false
+ *   eql?(other) -> true or false
+ *
+ *  Two structs of the same class and with the same content (according to Object#==)
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply struct.c]: Done (I think).

File: struct.c (hunk 1 above)
Comment (BurdetteLamar) [reply struct.c]: Done.

File: struct.c (hunk 2 above)
Comment (BurdetteLamar) [reply struct.c]: Done.

File: struct.c (hunk 3 above)
Comment (BurdetteLamar) [reply struct.c]: Done.

**Review by jeremyevans — APPROVED** [review]
Looks good.  Please wait until CI passes before merging this, as a non-comment brace was removed and then added back.

**Pull request merged**